src/chebyshev.jl

```diff
@@ -8,28 +8,58 @@ reccurence_deno_coef(::Type{<:AbstractChebyshevBasis}, degree) = 1
 
 """
     struct ChebyshevBasisFirstKind{P} <: AbstractChebyshevBasis{P}
-        polynomials::Vector{P}
+        elements::Vector{P}
     end
 
 Orthogonal polynomial with respect to the univariate weight function ``w(x) = \\frac{1}{\\sqrt{1 - x^2}}`` over the interval ``[-1, 1]``.
 """
 struct ChebyshevBasisFirstKind{P} <: AbstractChebyshevBasis{P}
-    polynomials::Vector{P}
+    elements::Vector{P}
 end
 
 const ChebyshevBasis{P} = ChebyshevBasisFirstKind{P}
 
 degree_one_univariate_polynomial(::Type{<:ChebyshevBasisFirstKind}, variable::MP.AbstractVariable) = MA.@rewrite(variable + 0)
 
+function scalar_product_function(::Type{<:ChebyshevBasisFirstKind})
+    function sp(i::Int)
+        if i == 0
+            return π
+        elseif mod(i, 2) == 1
+            return 0
+        else
+            n = Int(i/2)
+            return π*factorial(i)/(2^i * factorial(n)^2)
```

Comment:
This will very quickly exceed the capacity of `Int64`. We can simplify `factorial(i) with `factorial(n)`, do `pi / 2^i` so that it transforms to `Float64` before multiplying with `factorial(n)`